component/src/main/java/com/celements/blog/article/Article.java

```diff
@@ -441,8 +488,124 @@ private XWikiContext getContext() {
+    }
+    return articleExtract;
+  }
+
+  List<ImageUrl> getArticleImagesBySizeAsc(String lang) {
+    List<ImageUrl> articleImages = getImageUrlExtractor().extractImagesSocialMediaUrlList(
+        getFullArticle(lang));
+    if (articleImages.isEmpty()) {
+      articleImages = getImageUrlExtractor().extractImagesSocialMediaUrlList(getExtract(lang, false,
+          getMaxNumChars()));
+    }
+    return articleImages;
+  }
+
+  public @NotNull String getExternalUrl() {
+    String fullName = getModelUtils().serializeRefLocal(articleDocRef);
+    try {
+      return context.getWiki().getExternalURL(fullName, "view", "ref=socialmedia", context);
+    } catch (XWikiException xwe) {
+      LOGGER.error("Exception while trying to get externalURL for doc {}", articleDocRef, xwe);
+    }
+    return context.getWiki().getURL(articleDocRef, "view", context);
+  }
+
+  /**
+   * Returns social media meta tags, respecting the configuration. If the tags are activate the
+   * calculated OpenGraph tags are added. Furthermore if a Twitter account is configured
+   * (e.g. @example), the Twitter tags are added. The Twitter card type is configurable, defaulting
+   * to "summary"
+   *
+   * @param language
+   */
+  public @NotNull List<MetaTag> getArticleSocialMediaTags(@NotNull String language) {
+    List<MetaTag> metaTags = new ArrayList<>();
+    if (getConfigurationSource().getProperty(BLOG_ARTICLE_SOCIAL_MEDIA_CONF_NAME, false)) {
+      List<ImageUrl> images = getArticleImagesBySizeAsc(language);
+      metaTags.addAll(addOpenGraphTags(images, language));
+      metaTags.addAll(addTwitterTags(images));
+    }
+    return metaTags;
+  }
+
+  List<MetaTag> addOpenGraphTags(List<ImageUrl> images, String language) {
+    List<MetaTag> metaTags = new ArrayList<>();
+    String externalUrl = getExternalUrl();
+    metaTags.add(new MetaTag(EOpenGraph.OPENGRAPH_TYPE, "website"));
+    for (ImageUrl image : images) {
+      metaTags.add(new MetaTag(EOpenGraph.OPENGRAPH_IMAGE, image.getUrl()));
+      if (image.getWidth().isPresent()) {
+        metaTags.add(new MetaTag(EOpenGraph.OPENGRAPH_OPTIONAL_IMAGE_WIDTH,
+            image.getWidth().get().toString()));
+      }
+      if (image.getHeight().isPresent()) {
+        metaTags.add(new MetaTag(EOpenGraph.OPENGRAPH_OPTIONAL_IMAGE_HEIGHT,
+            image.getHeight().get().toString()));
+      }
+    }
+    metaTags.add(new MetaTag(EOpenGraph.OPENGRAPH_URL, externalUrl));
+    metaTags.add(new MetaTag(EOpenGraph.OPENGRAPH_TITLE, getTitleWithMenuNameFallback(
+        language).replaceAll("\"", "&quot;")));
+    // maxNumChars: e.g. for Facebook posts 300, for Facebook comments 110
+    // viewTypeFull: maxNumChars has no influence if viewTypeFull == true
+    String plainExtract = getExtractPlainTextEncoded(language, false, 450);
+    metaTags.add(new MetaTag(EOpenGraph.OPENGRAPH_OPTIONAL_DESCRIPTION, plainExtract));
+    return metaTags;
+  }
+
+  List<MetaTag> addTwitterTags(List<ImageUrl> images) {
```

Comment:
this method doesn't add tags, perhaps a better verb for naming would be "collect"?